src/TensorTwoD.cpp

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (c) 2015 - 2022, Intel Corporation
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+
+#include "config.h"
+#include "TensorTwoD.hpp"
+
+#include <cmath>
+#include <algorithm>
+#include <functional>
+#include <numeric>
+
+#include "geopm/Exception.hpp"
+#include "TensorOneD.hpp"
+
+namespace geopm
+{
+
+
+    TensorTwoD::TensorTwoD(std::size_t rows, std::size_t cols)
+    {
+        set_dim(rows, cols);
+    }
+
+    TensorTwoD::TensorTwoD(const TensorTwoD &other)
+        : m_mat(std::move(other.m_mat))
+    {
+    }
+
+    TensorTwoD::TensorTwoD(json11::Json input)
+    {
+
+        if (!input.is_array()){
+            throw geopm::Exception("Neural network weights is non-array-type.\n",
+                                   GEOPM_ERROR_INVALID, __FILE__, __LINE__);
+        }
+        if (input.array_items().size() == 0) {
+            throw geopm::Exception("Empty array is invalid for neural network weights.\n",
+                                   GEOPM_ERROR_INVALID, __FILE__, __LINE__);
+        }
+
+        set_dim(input.array_items().size(), input[0].array_items().size());
```

Comment:
Prior to this, check ``input[0].is_array()``. I see you are checking this in the ``for`` loop, but it'll need to be checked prior to the first access.

Reply:
Fixed